GitVersionCore/LibGitExtensions.cs

```diff
@@ -24,35 +24,48 @@ public static Branch FindBranch(this IRepository repository, string branchName)
             return repository.Branches.FirstOrDefault(x => x.Name == "origin/" + branchName);
         }
 
-        public static IEnumerable<Tag> TagsByDate(this IRepository repository, Commit commit)
+        public static Commit FindCommitBranchWasBranchedFrom(this Branch branch, IRepository repository, bool onlyTrackedBranches, params Branch[] excludedBranches)
         {
-            return repository.Tags
-                .Where(tag => tag.PeeledTarget() == commit)
-                .OrderByDescending(tag =>
-                {
-                    if (tag.Annotation != null)
-                    {
-                        return tag.Annotation.Tagger.When;
-                    }
-                    //lightweight tags will not have an Annotation
-                    return commit.Committer.When;
-                });
+            var currentBranches = branch.Tip.GetBranchesContainingCommit(repository, onlyTrackedBranches).ToList();
+            var tips = repository.Branches.Except(excludedBranches).Where(b => b != branch && !b.IsRemote).Select(b => b.Tip).ToList();
+            var branchPoint = branch.Commits.FirstOrDefault(c =>
+            {
+                if (tips.Contains(c)) return true;
+                var branchesContainingCommit = c.GetBranchesContainingCommit(repository, onlyTrackedBranches).ToList();
+                return branchesContainingCommit.Count > currentBranches.Count;
+            });
+            return branchPoint ?? branch.Tip;
         }
 
-        public static IEnumerable<Tag> SemVerTagsRelatedToVersion(this IRepository repository, Config configuration, SemanticVersion version)
+        public static IEnumerable<Branch> GetBranchesContainingCommit(this Commit commit, IRepository repository, bool onlyTrackedBranches)
```

Comment:
This will return references that point to a commit which is, or leads to (through parent-child relationships), the searched commit.